### PR TITLE
Overhaul of buildiso documentation and help messages

### LINKED ISSUE #2978: Buildiso: `distro` parameter is mandatory and not optional
### Describe the bug

When using the `buildiso` command, the help states that the `distro` parameter is optional but it is in fact mandatory:

```bash
$ cobbler buildiso --help
Usage: cobbler buildiso [options]

Options:
  -h, --help            show this help message and exit
  --iso=ISO             (OPTIONAL) output ISO to this file
  --profiles=PROFILES   (OPTIONAL) use these profiles only
  --systems=SYSTEMS     (OPTIONAL) use these systems only
  --tempdir=BUILDISODIR
                        (OPTIONAL) working directory
  --distro=DISTRO       (OPTIONAL) used with --standalone and --airgapped to
                        create a distro-based ISO including all associated
                        profiles/systems
  --standalone          (OPTIONAL) creates a standalone ISO with all required
                        distro files, but without any added repos
  --airgapped           (OPTIONAL) creates a standalone ISO with all distro
                        and repo files for disconnected system installation
  --source=SOURCE       (OPTIONAL) used with --standalone to specify a source
                        for the distribution files
  --exclude-dns         (OPTIONAL) prevents addition of name server addresses
                        to the kernel boot options
  --mkisofs-opts=MKISOFS_OPTS
                        (OPTIONAL) extra options for mkisofs
```

```bash
$ cobbler distro list
   testdistro

$ cobbler buildiso
task started: 2022-03-15_164432_buildiso
task started (id=Build Iso, time=Tue Mar 15 16:44:32 2022)
running python triggers from /var/lib/cobbler/triggers/task/buildiso/pre/*
running shell triggers from /var/lib/cobbler/triggers/task/buildiso/pre/*
shell triggers finished successfully
Exception occurred: <class 'ValueError'>
Exception value: Not existent distribution name passed to "cobbler buildiso"!
Exception Info:
!!! TASK FAILED !!!

$ cobbler buildiso testdistro
task started: 2022-03-15_164428_buildiso
task started (id=Build Iso, time=Tue Mar 15 16:44:28 2022)
running python triggers from /var/lib/cobbler/triggers/task/buildiso/pre/*
running shell triggers from /var/lib/cobbler/triggers/task/buildiso/pre/*
shell triggers finished successfully
Exception occurred: <class 'ValueError'>
Exception value: Not existent distribution name passed to "cobbler buildiso"!
Exception Info:
!!! TASK FAILED !!!
```

### Steps to reproduce

1. Run `cobbler buildiso`
2. See that the optional `distro` parameter is not optional

### Expected behavior

It should be clear that the `distro` parameter is not optional but mandatory.

### Cobbler version

````bash
cobbler version
Cobbler 3.3.2
  source: ?, ?
  build time: Tue Mar 15 12:00:00 2022

````

### Operating system

```bash
$ cat /etc/os-release 
NAME="SLES"
VERSION="15-SP4"
VERSION_ID="15.4"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP4"
ID="sles"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:15:sp4"
DOCUMENTATION_URL="https://documentation.suse.com/"
```

### Cobbler log

```log
[Thread-127] 2022-03-15T16:44:32 - INFO | REMOTE start_task(Build Iso); event_id(2022-03-15_164432_buildiso); user(?)
[2022-03-15_164432_buildiso] 2022-03-15T16:44:33 - DEBUG | running python triggers from /var/lib/cobbler/triggers/task/buildiso/pre/*
[2022-03-15_164432_buildiso] 2022-03-15T16:44:33 - DEBUG | running shell triggers from /var/lib/cobbler/triggers/task/buildiso/pre/*[2022-03-15_164432_buildiso] 2022-03-15T16:44:33 - DEBUG | shell triggers finished successfully
[2022-03-15_164432_buildiso] 2022-03-15T16:44:33 - INFO | Exception occurred: <class 'ValueError'>[2022-03-15_164432_buildiso] 2022-03-15T16:44:33 - INFO | Exception value: Not existent distribution name passed to "cobbler buildiso"![2022-03-15_164432_buildiso] 2022-03-15T16:44:33 - INFO | Exception Info:
  File "/usr/lib/python3.6/site-packages/cobbler/remote.py", line 99, in run 
    rc = self._run(self) 
  File "/usr/lib/python3.6/site-packages/cobbler/remote.py", line 174, in runner
    self.options.get("xorrisofs_opts", ""),
  File "/usr/lib/python3.6/site-packages/cobbler/api.py", line 1769, in build_iso    distro_name=distro_name, systems=systems, exclude_dns=exclude_dns)
  File "/usr/lib/python3.6/site-packages/cobbler/actions/buildiso/netboot.py", line 565, in run
    buildisodir = self._prepare_iso(buildisodir, distro_name, profiles)
  File "/usr/lib/python3.6/site-packages/cobbler/actions/buildiso/__init__.py", line 289, in _prepare_iso
    ) from value_error
[2022-03-15_164432_buildiso] 2022-03-15T16:44:33 - ERROR | ### TASK FAILED ###
```

### LINKED ISSUE #2979: [docs] Outdated migration matrix
The migration matrix found [here](https://cobbler.readthedocs.io/en/latest/cobbler-conf.html#migration-matrix) is outdated and should be updated for 3.3.1 and 3.3.2.

### LINKED ISSUE #2983: Buildiso: Better explanation of `--standalone` and `--airgapped`
### Describe the bug

These are several point but all related to the same issue so I did not split them up in different ones.

1. The `buildiso` help should be extended to include the fact that the `--standalone` and `--airgapped` parameter must be supplied with the `--source` parameter. 
2. According to [Cobbler buildiso](https://cobbler.readthedocs.io/en/latest/cobbler.html?#cobbler-buildiso) it is not clear that this must be done. On the other side we have [Building standalone ISOs](https://cobbler.readthedocs.io/en/latest/user-guide/building-isos.html#building-standalone-isos) with remarks on this.
3. After reading the above docs, to me it is not clear what `--standalone` and `--airgapped` actually do and what the difference of both is?
4. The error shown after running the command is exactly the same for `--standalone` as for `--airgapped`: 
  `Exception value: Error, no installation source found. When building a standalone ISO, you must specify a --source if the distro install tree is not hosted locally` The message should be adjusted according to what parameter was used.

```bash
cobbler buildiso --help
Usage: cobbler buildiso [options]

Options:
  -h, --help            show this help message and exit
  --iso=ISO             (OPTIONAL) output ISO to this file
  --profiles=PROFILES   (OPTIONAL) use these profiles only
  --systems=SYSTEMS     (OPTIONAL) use these systems only
  --tempdir=BUILDISODIR
                        (OPTIONAL) working directory
  --distro=DISTRO       (OPTIONAL) used with --standalone and --airgapped to
                        create a distro-based ISO including all associated
                        profiles/systems
  --standalone          (OPTIONAL) creates a standalone ISO with all required
                        distro files, but without any added repos
  --airgapped           (OPTIONAL) creates a standalone ISO with all distro
                        and repo files for disconnected system installation
  --source=SOURCE       (OPTIONAL) used with --standalone to specify a source
                        for the distribution files
  --exclude-dns         (OPTIONAL) prevents addition of name server addresses
                        to the kernel boot options
  --mkisofs-opts=MKISOFS_OPTS
                        (OPTIONAL) extra options for mkisofs
```

```bash
$ cobbler buildiso --iso=/var/cache/cobbler/autoinstall.iso --distro=testdistro --airgapped
task started: 2022-03-15_201836_buildiso
task started (id=Build Iso, time=Tue Mar 15 20:18:36 2022)
running python triggers from /var/lib/cobbler/triggers/task/buildiso/pre/*
running shell triggers from /var/lib/cobbler/triggers/task/buildiso/pre/*
shell triggers finished successfully
Deleting and recreating the buildisodir at "/var/cache/cobbler/buildiso"
building tree for isolinux
building tree for GRUB 2
copying syslinux files
copying: /usr/share/syslinux/isolinux.bin -> /var/cache/cobbler/buildiso/isolinux/isolinux.bin
copying: /usr/share/syslinux/menu.c32 -> /var/cache/cobbler/buildiso/isolinux/menu.c32
copying: /usr/share/syslinux/chain.c32 -> /var/cache/cobbler/buildiso/isolinux/chain.c32
copying GRUB2 files
copying: /var/lib/cobbler/loaders/grub/grubx86.efi -> /var/cache/cobbler/buildiso/grub/grub.efi
Trying to locate source for distro
Exception occurred: <class 'ValueError'>
Exception value: Error, no installation source found. When building a standalone ISO, you must specify a --source if the distro install tree is not hosted locally
Exception Info:
!!! TASK FAILED !!!
```

### Steps to reproduce

1. Run `cobbler buildiso --standalone` or `cobbler buildiso --airgapped` without `--source`
2. See the error message from above

### Expected behavior

It should be clear that the parameter `--source` is mandatory when using the parameters `--standalone`/`--airgapped`. Furthermore it should be clear what both exactly do and what the difference between them is.

### Cobbler version

````
$ cobbler version
Cobbler 3.3.2
  source: ?, ?
  build time: Tue Mar 15 12:00:00 2022

````

### Operating system

```bash
cat /etc/os-release 
NAME="SLES"
VERSION="15-SP4"
VERSION_ID="15.4"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP4"
ID="sles"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:15:sp4"
DOCUMENTATION_URL="https://documentation.suse.com/"
```

### Cobbler log

````log
[Thread-506] 2022-03-15T20:04:58 - INFO | REMOTE start_task(Build Iso); event_id(2022-03-15_200458_buildiso); user(?)
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - DEBUG | running python triggers from /var/lib/cobbler/triggers/task/buildiso/pre/*
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - DEBUG | running shell triggers from /var/lib/cobbler/triggers/task/buildiso/pre/*
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - DEBUG | shell triggers finished successfully
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - INFO | Deleting and recreating the buildisodir at "/var/cache/cobbler/buildiso"
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - INFO | building tree for isolinux
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - INFO | building tree for GRUB 2
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - INFO | copying syslinux files
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - INFO | copying: /usr/share/syslinux/isolinux.bin -> /var/cache/cobbler/buildiso/isolinux/isolinux.bin
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - INFO | copying: /usr/share/syslinux/menu.c32 -> /var/cache/cobbler/buildiso/isolinux/menu.c32
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - INFO | copying: /usr/share/syslinux/chain.c32 -> /var/cache/cobbler/buildiso/isolinux/chain.c32
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - INFO | copying GRUB2 files
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - INFO | copying: /var/lib/cobbler/loaders/grub/grubx86.efi -> /var/cache/cobbler/buildiso/grub/grub.efi
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - DEBUG | Trying to locate source for distro
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - INFO | Exception occurred: <class 'ValueError'>
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - INFO | Exception value: Error, no installation source found. When building a standalone ISO, you must specify a --source if the distro install tree is not hosted locally
[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - INFO | Exception Info:
  File "/usr/lib/python3.6/site-packages/cobbler/remote.py", line 99, in run
    rc = self._run(self)

  File "/usr/lib/python3.6/site-packages/cobbler/remote.py", line 174, in runner
    self.options.get("xorrisofs_opts", ""),

  File "/usr/lib/python3.6/site-packages/cobbler/api.py", line 1765, in build_iso
    distro_name=distro_name, airgapped=airgapped, source=source)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/buildiso/standalone.py", line 344, in run
    self._validate_standalone_args(distro_name, source)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/buildiso/standalone.py", line 64, in _validate_standalone_args
    source = self._validate_standalone_filesource(source, base_distro)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/buildiso/standalone.py", line 128, in _validate_standalone_filesource
    "Error, no installation source found. When building a standalone ISO, you must specify a "

[2022-03-15_200458_buildiso] 2022-03-15T20:04:59 - ERROR | ### TASK FAILED ###
[Thread-513] 2022-03-15T20:05:17 - DEBUG | REMOTE CLI Authorized; user(?)
[Thread-513] 2022-03-15T20:05:17 - INFO | REMOTE start_task(Build Iso); event_id(2022-03-15_200517_buildiso); user(?)
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - DEBUG | running python triggers from /var/lib/cobbler/triggers/task/buildiso/pre/*
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - DEBUG | running shell triggers from /var/lib/cobbler/triggers/task/buildiso/pre/*
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - DEBUG | shell triggers finished successfully
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - INFO | Deleting and recreating the buildisodir at "/var/cache/cobbler/buildiso"
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - INFO | building tree for isolinux
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - INFO | building tree for GRUB 2
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - INFO | copying syslinux files
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - INFO | copying: /usr/share/syslinux/isolinux.bin -> /var/cache/cobbler/buildiso/isolinux/isolinux.bin
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - INFO | copying: /usr/share/syslinux/menu.c32 -> /var/cache/cobbler/buildiso/isolinux/menu.c32
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - INFO | copying: /usr/share/syslinux/chain.c32 -> /var/cache/cobbler/buildiso/isolinux/chain.c32
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - INFO | copying GRUB2 files
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - INFO | copying: /var/lib/cobbler/loaders/grub/grubx86.efi -> /var/cache/cobbler/buildiso/grub/grub.efi
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - DEBUG | Trying to locate source for distro
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - INFO | Exception occurred: <class 'ValueError'>
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - INFO | Exception value: Error, no installation source found. When building a standalone ISO, you must specify a --source if the distro install tree is not hosted locally
[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - INFO | Exception Info:
  File "/usr/lib/python3.6/site-packages/cobbler/remote.py", line 99, in run
    rc = self._run(self)

  File "/usr/lib/python3.6/site-packages/cobbler/remote.py", line 174, in runner
    self.options.get("xorrisofs_opts", ""),

  File "/usr/lib/python3.6/site-packages/cobbler/api.py", line 1765, in build_iso
    distro_name=distro_name, airgapped=airgapped, source=source)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/buildiso/standalone.py", line 344, in run
    self._validate_standalone_args(distro_name, source)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/buildiso/standalone.py", line 64, in _validate_standalone_args
    source = self._validate_standalone_filesource(source, base_distro)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/buildiso/standalone.py", line 128, in _validate_standalone_filesource
    "Error, no installation source found. When building a standalone ISO, you must specify a "

[2022-03-15_200517_buildiso] 2022-03-15T20:05:18 - ERROR | ### TASK FAILED ###
````

### PR DESCRIPTION
**Pull request by SchoolGuy**
Fixes #2978
Fixes #2983
Fixes #2979

Commit message should contain all the details required for this PR.

### CODE REVIEW

**Comment by SchoolGuy** [comment]
@nodeg I would appreciate if you could check if the documentation and help messages are now better or if you need further clarification. I am unsure what more to write as I have never tried to actually use a produced image.

File: cobbler/cli.py [hunk 1]
```diff
@@ -2424,26 +2422,23 @@ def direct_command(self, action_name: str):
                 "--airgapped",
                 dest="airgapped",
                 action="store_true",
-                help="(OPTIONAL) creates a standalone ISO with all distro and repo files for "
-                "disconnected system installation",
+                help="(OPTIONAL) implies --standalone but additionally includes the repositories files in the ISO",
```
Comment (nodeg) [inline cobbler/cli.py:2425]:
```suggestion
                help="(OPTIONAL) implies --standalone but additionally includes the repository files into ISO",
```

File: docs/cobbler.rst [hunk 2]
```diff
@@ -944,28 +944,31 @@ dependencies and configuration. The in depth explanation can be found at :ref:`b
 +--------------+-------------------------------------------------------------------------------------------------------+
 | profiles     | Use these profiles only for information collection.                                                   |
 +--------------+-------------------------------------------------------------------------------------------------------+
-| systems      | Use these systems only for information collection.                                                    |
+| systems      | (net-only) Use these systems only for information collection.                                         |
```
Comment (nodeg) [inline docs/cobbler.rst:947]: Is this really net-only?

**Comment by SchoolGuy** [comment]
@nodeg Since we are stuck with `optparse` and my desire to push the new CLI, I wont fix the usage. In `optparse` It is [not possible to do what you propose easily](https://docs.python.org/3/library/optparse.html#background). Thus I will try to fix your concerns only in the understanding of what you can read and not the behavior.

**Comment by nodeg** [comment]
Fine with me. We already have the issue open for the switch from optparse to argparse so I will just add a note there.

File: cobbler/cli.py [hunk 3]
```diff
@@ -2409,9 +2409,7 @@ def direct_command(self, action_name: str):
             self.parser.add_option(
                 "--distro",
                 dest="distro",
-                help="(OPTIONAL) used with --standalone and --airgapped "
-                "to create a distro-based ISO including all "
-                "associated profiles/systems",
+                help="Specify this to choose the Kernel and Initrd for the ISO being built.",
```
[the remark this answers is not in the document]
Comment (SchoolGuy) [reply cobbler/cli.py:2412]: What do you think of the following?

```suggestion
                help="Must be specified to choose the Kernel and Initrd for the ISO being built.",
```

**Review by nodeg — APPROVED** [review]
LGTM. Thank you for the quick fixes!

**Pull request merged**